### PR TITLE
add frontend numpy API: ndarray.__array__ - Close #10249

### LINKED ISSUE #10249: __array__

### PR DESCRIPTION
**Pull request by guodh321**
Hi, I made an initial solution to this issue.

I noticed the issue I created hadn't been added to the to-do list here - https://github.com/unifyai/ivy/issues/3607. 

Could you help me add it to the to-do list first and then have a look at the solution in this pull request - if it's accepted or needs further change?

Thanks!

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 1]
```diff
@@ -2246,3 +2246,33 @@ def test_numpy_instance_len__(
...
+# __array__
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="__array__",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+    ),
+)
+def test_numpy_instance_array__(
```
Comment (ZoeCD) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2259]: LGTM! Please resolve the conflicting files, and the PR will be ready to merge! Thank you for your contribution 😊

**Review by ZoeCD — CHANGES_REQUESTED** [review]

**Review by ZoeCD — APPROVED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 1 above)
Comment (danny-1k) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2349]: @ZoeCD,  @guodh321 Line `2259` of ```ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py``` does not look right.

Comment (guodh321) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2349]: Hi @danny-1k , I found some lines got lost when merging into the master, could you or your colleagues add these below? Thanks.
    dtype_and_x,
    frontend_method_data,
    init_flags,
    method_flags,
    frontend,
):
    input_dtypes, x = dtype_and_x
    helpers.test_frontend_method(
        init_input_dtypes=input_dtypes,
        init_all_as_kwargs_np={
            "object": x[0],
        },
        method_input_dtypes=input_dtypes,
        method_all_as_kwargs_np={},
        init_flags=init_flags,
        method_flags=method_flags,
        frontend=frontend,
        frontend_method_data=frontend_method_data,
    )

**Pull request merged**